### PR TITLE
download media from tg links #441

### PR DESCRIPTION
**Pull request by Krishna-Singhal**

### CODE REVIEW

File: userge/plugins/misc/download.py [hunk 1]
```diff
@@ -72,6 +73,24 @@ async def handle_download(message: Message, resource: Union[Message, str]) -> Tu
...
 async def url_download(message: Message, url: str) -> Tuple[str, int]:
     """ download from link """
+    pattern = r"^(?:https?:\/\/)?(?:www\.)?t\.me\/(?:(?:c\/(\d+))|(\w+))\/(\d+)$"
+    # group 1: private chat id, group 2: chat username, group 3: message id
+    match = re.search(pattern, url.split('|', 1)[0].strip())
+    if match:
+        chat_id = None
+        msg_id = int(match.group(3))
+        if match.group(1):
+            chat_id = int("-100" + match.group(1))
```
Comment (rking32) [inline userge/plugins/misc/download.py:83]: I'm not sure about this line. I guess, sometimes we should add -100 and sometimes not. Need some tests though.

Comment (Krishna-Singhal) [reply userge/plugins/misc/download.py:83]: didn't find any case where shouldn't not use -100 before chat id if chat is private

Comment (rking32) [reply userge/plugins/misc/download.py:83]: ok, then good

**Comment by rking32** [comment]
Thanks, merging ...

**Pull request merged**